src/app/util/attribute-storage.cpp

```diff
@@ -347,7 +427,7 @@ CHIP_ERROR emberAfSetDynamicEndpointWithEpUniqueId(uint16_t index, EndpointId id
 
     MutableCharSpan targetSpan(emAfEndpoints[index].endpointUniqueId);
     CopyCharSpanToMutableCharSpanWithTruncation(endpointUniqueId, targetSpan);
-    emAfEndpoints[index].endpointUniqueIdSize = targetSpan.size();
+    emAfEndpoints[index].endpointUniqueIdSize = static_cast<uint8_t>(std::min<size_t>(targetSpan.size(), UINT8_MAX));
```

Comment:
Why do we need this min business? The size is not going to be greater than `sizeof(emAfEndpoints[index].endpointUniqueId)`, so we could just static_assert that fits in uint8_t?